src/plugins/discover/public/plugin.tsx

```diff
@@ -320,22 +331,30 @@ export class DiscoverPlugin
     return { rootProfileService, dataSourceProfileService, documentProfileService };
   });
 
-  private async createProfilesManager({ plugins }: { plugins: DiscoverStartPlugins }) {
+  private async createProfilesManager({
+    plugins,
+    ebtContextManager,
+  }: {
+    plugins: DiscoverStartPlugins;
+    ebtContextManager: DiscoverEBTContextManager;
+  }) {
     const { rootProfileService, dataSourceProfileService, documentProfileService } =
       await this.createProfileServices({ plugins });
 
     return new ProfilesManager(
       rootProfileService,
       dataSourceProfileService,
-      documentProfileService
+      documentProfileService,
+      ebtContextManager
     );
   }
 
   private createEmptyProfilesManager() {
     return new ProfilesManager(
       new RootProfileService(),
       new DataSourceProfileService(),
-      new DocumentProfileService()
+      new DocumentProfileService(),
+      new DiscoverEBTContextManager() // it's not enabled outside of Discover
```

Comment:
I sort of overlooked that this approach wouldn't work for embeddables in our initial discussions about it. I think it works well for events in Discover and we should keep it, but I also feel like we may need an additional approach later not based around global EBT context to help track events in dashboard panels. This still seems like a great place to start though since Discover usage is most important right now.